develop-guide/writeparser/writeparser.rst

```diff
@@ -1,47 +1,558 @@
+|   5.2 :ref:`Explore other pyATS schemas <schemaexamples>`
+| 6. :ref:`Writing the Parser Class <writing_the_parser_class>`
+
+TODO
+Finish this ^^ table of contents
+
+.. _what_is_a_parser:
+
+********************
+1. What is a parser?
+********************
+
+.. include:: ../definitions/def_parser.rst
+    :start-line: 3
+
+* For a basic introduction to the |library| parsers, see the topic `Parse device output <https://pubhub.devnetcloud.com/media/pyats-getting-started/docs/quickstart/parseoutput.html#parse-device-output>`_ in the *Get Started with pyATS* guide (recommended).
+* For more information about the ``metaparser`` package, see the topic `Metaparser Package <https://pubhub.devnetcloud.com/media/genie-metaparser/docs/index.html#genie-metaparser-package>`_.
+
+
+.. _why_write_a_parser:
+
+**********************
+2. Why write a parser?
+**********************
+
+The |library| provides you with many out-of-the box `parsers <https://pubhub.devnetcloud.com/media/genie-feature-browser/docs/#/parsers>`_ for use with the most frequently-used Cisco show commands and OS/platform combinations.
+
+But if you want to customize a parser, make a new parser, or if you need to parse output for a feature that does not yet have a |library| model, you can easily write your own parser.
+
+
+.. _how_the_parsers_work:
+
+*********************************
+3. How the |library| parsers work
+*********************************
+
+A parser is composed of two Python classes:
+
+- :ref:`Schema class <writing_the_schema_class>`
+- :ref:`Parser class <writing_the_parser_class>`
+
+| Each schema defines:
+|   1. What the structure of our parsed data will be.
+|   2. How that structure is composed through the use of key-value pairs.
+
+| Each parser performs four main actions:
+|   1. Run a show command and collect the device output.
+|   2. Define patterns that will be looked for in the device output.
+|   3. Iterate through each line of the device output data.
+|   4. Look for patterns in the output and add any matching patterns to a Python dictionary.
+
+The result of the parser and schema classes working together is a standardized
+and *structured* output that works with network automation scripts, regardless
+of OS or communication protocol.
+
+Later sections will describe this process in more detail and multiple examples
+will be provided.
+
+
+.. _setting_up_your_environment:
+
+******************************************
+4. Setting up your development environment
+******************************************
+
+If this is your first time writing a pyATS parser or if you're looking for an easy way
+to jump right into things, then start here. This section will outline how to
+easily set up a development environment that can be used for writing a parser.
+
+Things you'll need before starting:
+
+- Python 3.6 (or newer)
+- pip
+- git
+
+#. Fork the https://github.com/CiscoTestAutomation/genieparser repo.

```

Comment:
can you add a picture on how to fork the repo. Screenshot put red arrow on the button  or something like this.

Reply:
Added a screenshot